ssh/tailssh/incubator_linux.go

```diff
@@ -0,0 +1,170 @@
+// Copyright (c) 2022 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+//go:build linux
+// +build linux
+
+package tailssh
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"syscall"
+	"time"
+	"unsafe"
+
+	"github.com/godbus/dbus/v5"
+	"tailscale.com/types/logger"
+)
+
+func init() {
+	ptyName = ptyNameLinux
+	startLoginSession = startLoginSessionLinux
+}
+
+func ptyNameLinux(f *os.File) (string, error) {
+	var n uint32
+	_, _, e := syscall.Syscall(syscall.SYS_IOCTL, f.Fd(), syscall.TIOCGPTN, uintptr(unsafe.Pointer(&n)))
+	if e != 0 {
+		return "", e
+	}
+	return fmt.Sprintf("pts/%d", n), nil
+}
+
+// callLogin1 invokes the provided method over D-Bus.
```

Comment:
.... provided method of the "login1" service over ...

(to explain why this method is named as such)